magento/inventory
# MSI#822 Source deduction for virtual and downloadable products

### Description
Source Deduction logic for virtual and downloadable products.

### Fixed Issues (if relevant)

1. magento-engcom/msi#822: Downloadable and Virtual products remain In Stock after all quantity is bought.

### Manual testing scenarios

1. Create Virtual Product while using Single Stock Mode
2. Buy the virtual product 
3. (Database check) 
     * Reservation was created
     * Source items are not decreased yet
4. Go to `Sales > Orders` in admin 
5. Select your order and create an invoice
6. (Database check) 
     * A new Reservation entry was created, which negotiates the first initial reservation
     * Source items are deducted by the invoiced quantity
7. Create further sources and a new stock; assign your website to the new stock
8. Assign the new sources to your virtual product and add some quantity
9. Repeat order and invoicing and check the expected changes in database

### Testing variations
* Process an order with mixed product types (virtual, downloadable, simple, etc.)
* Try partial invoicing (@maghamed, @naydav: This could probably cause an issue with reservation cleanup...?)

### Contribution checklist
